### PR TITLE
single use tool 

### PR DESCRIPTION
**Pull request by Glease**
close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/14102

recipes and balance are subject to discussions. Currently there is no matching multiblock for this except PA, but the sheer throughput from HSS-G should invalidate any need of parallelizing/OC. I intend on adding a smart multiblock for this (can respond to ME crafting request dynamically) once first fully functioning MuTE multiblock is out.

MV
image

HV
image

EV
image

shapes looks like this. neither tool used will not get consumed. They will only get weared out slightly just like any other crafting. I can't say this is a very brilliant recipe, but IMO it's enough as a placeholder until someone can come up with more interesting ones.
image
image

Currently the count of tools crafted is the durability of each tool with given tool material divided by durability cost per craft. The only exception is silver (which is added purely as a meme), whose durability is halved during calculation. https://github.com/GTNewHorizons/GigaGramFab/pull/23/commits/79499aab82a72d1d180ca2ea2fdc8a8252d8c3fe introduced a slightly less stupid algorithm. As a result most of the tool recipes now require a multiple of 16 of fluids as input.

### CODE REVIEW

**Comment by bombcar** [comment]
Would forming press for the tool molds make sense?

**Comment by Glease** [comment]
Due to technical limitations, that'd either break the tool, or not harm it at all. neither is very desireable, though if it's imperative to bring this handcraft to a machine then yes I'd probably move it to forming press.

File: src/main/java/net/glease/ggfab/api/GG_RecipeMaps.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import gregtech.api.gui.modularui.GT_UITextures;
+import gregtech.api.util.GT_Recipe;
+import gregtech.api.util.GT_RecipeBuilder;
+
+public class GG_RecipeMaps {
+
+    public static final GT_RecipeBuilder.MetadataIdentifier<ToolDictNames> OUTPUT_TYPE = GT_RecipeBuilder.MetadataIdentifier
+            .create(ToolDictNames.class, "output_type");
+    public static final GT_RecipeBuilder.MetadataIdentifier<Integer> OUTPUT_COUNT = GT_RecipeBuilder.MetadataIdentifier
+            .create(Integer.class, "output_COUNT");
```
Comment (miozune) [inline src/main/java/net/glease/ggfab/api/GG_RecipeMaps.java:25]: Is it intended "count" being uppercase?

File: src/main/java/net/glease/ggfab/SingleUseToolRecipeLoader.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            } else if (count < 128) {
+                long mod = GT_Utility.ceilDiv(128, count);
+                fluids *= mod;
+                duration *= mod;
+                count *= mod;
+            }
+            GT_Values.RA.stdBuilder().fluidInputs(material.getMolten(fluids)) //
+                    .metadata(GG_RecipeMaps.OUTPUT_TYPE, type) //
+                    .metadata(GG_RecipeMaps.OUTPUT_COUNT, (int) count) //
+                    .eut(TierEU.RECIPE_MV).duration(duration) //
```
Comment (miozune) [inline src/main/java/net/glease/ggfab/SingleUseToolRecipeLoader.java:95]: You might already know but you can update `blowdryerSetup` in `settings.gradle` to use GT5 styled spotless format, but maybe not in this PR

**Review by miozune — COMMENTED** [review]
BTW why are you using GG as acronym of GigaGramFab? It clashes with GoodGenerator

File: src/main/java/net/glease/ggfab/api/GG_RecipeMaps.java (hunk 1 above)
Comment (Glease) [reply src/main/java/net/glease/ggfab/api/GG_RecipeMaps.java:25]: uh no. probably a fatfinger.

**Comment by Dream-Master** [comment]
miozune — Today at 1:30 PM
code is ok, but I won't say anything around balance 
DreamMasterXXL — Today at 1:30 PM
oh fair enough. will ask around

We can test on zeta and tweak it if balance isnt godd.

**Review by Dream-Master — APPROVED** [review]

**Pull request merged**